### PR TITLE
Email whern sign up

### PR DESCRIPTION
**Pull request by PaulTetaz**

### CODE REVIEW

File: app/controllers/friend_invitations_controller.rb [hunk 1]
```diff
@@ -2,12 +2,14 @@ class FriendInvitationsController < ApplicationController
...
   def new
     @invitation = FriendInvitation.new
+    UserMailer.creation_confirmation(@user).deliver_now
   end
 
   def create
     @invitation = FriendInvitation.new(invitation_params)
     @invitation.user = @user
     @invitation.save
+    UserMailer.creation_confirmation(@user).deliver_now
```
Comment (contatc2) [inline app/controllers/friend_invitations_controller.rb]: let's rediscuss the flow and understand when the e-mail needs to go out, probably only at the create step.

File: app/mailers/application_mailer.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 class ApplicationMailer < ActionMailer::Base
-  default from: 'from@example.com'
+  default from: 'paultetaz@yahoo.fr'
```
Comment (contatc2) [inline app/mailers/application_mailer.rb]: should we update that if we use postmark?

File: app/views/user_mailer/creation_confirmation.html.erb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+  </head>
+  <body> -->
+    <h1>Awesome <%= @user.first_name %>, you just added a friend</h1>
+    <p>
+      You have successfully signed up to the website,
+      your username is: <%= @user.last_name %>.<br>
+    </p>
+    <p>
+      To login to the site, click <a href="<%= @url %>" target="blank">here</a>
```
Comment (contatc2) [inline app/views/user_mailer/creation_confirmation.html.erb]: where is @url defined?

File: app/views/user_mailer/creation_confirmation.html.erb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<!-- <!DOCTYPE html>
+<html>
+  <head>
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+  </head>
+  <body> -->
+    <h1>Awesome <%= @user.first_name %>, you just added a friend</h1>
```
Comment (contatc2) [inline app/views/user_mailer/creation_confirmation.html.erb]: invited a friend

File: app/views/user_mailer/welcome_email.html.erb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+  </head>
+  <body>
+    <h1>Welcome to IRL, <%= @user.first_name %></h1>
+    <p>
+      You have successfully signed up to the website,
+      your username is: <%= @user.last_name %>.<br>
+    </p>
+    <p>
+      To login to the site, click <a href="<%= @url %>" target="blank">here</a>
```
Comment (contatc2) [inline app/views/user_mailer/welcome_email.html.erb]: where is the @url defined?

File: config/routes.rb [hunk 6]
```diff
@@ -5,7 +5,7 @@
   }
   root to: 'pages#home'
   resources :users, only: %i[index show update] do
-    resources :user_adjectives, only: %i[new create]
+    resources :user_adjectives, only: %i[index new create update]
```
Comment (contatc2) [inline config/routes.rb]: do we need index ?

File: test/mailers/previews/user_mailer_preview.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+# Preview all emails at http://localhost:3000/rails/mailers/user_mailer
+class UserMailerPreview < ActionMailer::Preview
+  def welcome_email
+    user = User.first
```
Comment (contatc2) [inline test/mailers/previews/user_mailer_preview.rb]: why do we send it to User.first only?

Comment (contatc2) [reply test/mailers/previews/user_mailer_preview.rb]: it's a test, sorry, doesn't matter

**Comment by contatc2** [comment]
can we have a look at my comment after lunch and I'll need a ticket to solve the gemfile.lock, i would keep everything, but would rather confirm with a TA.

File: config/routes.rb (hunk 6 above)
Comment (PaulTetaz) [reply config/routes.rb]: nop

**Pull request merged**